### PR TITLE
Add support for sequence type for h2

### PR DESCRIPTION
**Pull request by rlconst**
This is rough attempt to support sequence types for H2 http://www.h2database.com/html/grammar.html#column_definition

### CODE REVIEW

**Comment by dstepanov** [comment]
Looks like there are failing tests

**Comment by rlconst** [comment]
Rebased. Github doesn't see signed CLA for some reason

**Comment by rlconst** [comment]
Weird exception https://github.com/micronaut-projects/micronaut-data/pull/1789/checks#step:8:13024

File: data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy [hunk 1]
```diff
@@ -173,6 +173,50 @@ interface MyRepository {
...
+    Dialect dialect() default Dialect.ANSI;
+}
+''')
+
+        PersistentEntity entity = PersistentEntity.of(Sale)
+        QueryBuilder builder = new SqlQueryBuilder(Dialect.H2)
+        def queryModel = QueryModel.from(entity).eq("name", QueryParameter.of("name"))
+
+        expect:
+        builder.dialect == Dialect.H2
```
Comment (radovanradic) [inline data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy:213]: I think `buildQuery` method has been changed in 4.0 and need `AnnotationMetadata` so just need to add this `AnnotationMetadata.EMPTY_METADATA` as first parameter, for example.

Comment (rlconst) [reply data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy:213]: @radovanradic I fixed failing test

**Review by radovanradic — APPROVED** [review]

**Comment by dstepanov** [comment]
Please target 4.9

**Comment by sdelamo** [comment]
> Please target 4.9

4.8.x is fine. 4.8.x is the next major branch.

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**